### PR TITLE
Add autocomplete definitions for Tool component

### PR DESCRIPTION
**Pull request by C3pa**
I am open to suggestions if you have an idea how to better name those types.

### CODE REVIEW

File: Data Files/MWSE/mods/CraftingFramework/types/Tool.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---@meta
+
+---@class craftingFrameworkToolData
+---@field id string **Required.**  This will be the unique identifier used internally by Crafting Framework to identify this `tool`.
+---@field name string The name of the tool. Used in various UIs.
+---@field ids table<number, string> **Required.**  This is the list of game item ids that will be used as the tool.
+
+---@class craftingFrameworkToolRequirements
+---@field equipped boolean Perform a check whether the tool is equipped?
```
Comment (jhaakma) [inline Data Files/MWSE/mods/CraftingFramework/types/Tool.lua]: When true, room must be equipped to be considered valid.

This table also has a count field which determines how many of the items need to be in the players inventory

Comment (jhaakma) [reply Data Files/MWSE/mods/CraftingFramework/types/Tool.lua]: The full definition of thus requirements take can be found in Recipe.lua in the ToolRequirementsSchema

https://github.com/jhaakma/crafting-framework/blob/d0b62aace3a54de8bea2113f8cb8ea3522c3083a/Data%20Files/MWSE/mods/CraftingFramework/components/Recipe.lua#L17

**Review by jhaakma — CHANGES_REQUESTED** [review]

**Comment by C3pa** [comment]
I've added the missing fields. Also, I changed a few descriptions to make them more clear. Do you find them acceptable?

**Comment by C3pa** [comment]
I made functions accepting tables have each of the fields documented. It does mean that I had to duplicate the comments at every site where `craftingFrameworkToolRequirements` is used but gives nice autocomplete. Btw, this is now how we have it set up in MWSE's annotations. Here is the end product:

image

**Comment by C3pa** [comment]
Finally, I have found a way how to have the EmmyLua annotations, not in the code file. So, in the source file, we need to declare `---@class craftingFrameworkTool` at the line before the `Tool` table is created. This will connect the annotations to the source code. Then in the source file, we only need to specify the arguments and return type for functions/methods.  Then, in the meta file the functions have all the parameters with comments.

**Review by jhaakma — APPROVED** [review]

**Pull request merged**